vector/src/main/java/im/vector/app/features/home/room/detail/RoomDetailViewModel.kt

```diff
@@ -972,9 +978,29 @@ class RoomDetailViewModel @AssistedInject constructor(
                     visibleEventsObservable.accept(RoomDetailAction.TimelineEventTurnsVisible(event))
                 }
             }
+
+            // handle chat effects here
+            if (vectorPreferences.chatEffectsEnabled()) {
+                chatEffectManager.checkForEffect(action.event)
+            }
+        }
+    }
+
+    override fun shouldStartEffect(effect: ChatEffect) {
+        when (effect) {
+            ChatEffect.CONFETTI -> {
+                _viewEvents.post(RoomDetailViewEvents.StartChatEffect(ChatEffect.CONFETTI))
+            }
+            ChatEffect.SNOW -> {
+                _viewEvents.post(RoomDetailViewEvents.StartChatEffect(ChatEffect.SNOW))
+            }
```

Comment:
Replace the whole `when` block by `_viewEvents.post(RoomDetailViewEvents.StartChatEffect(effect))` ?

Reply:
Done